# Lab | Inferential statistics - T-test & P-value

## 1.  We will have another simple example on two sample t test (pooled- when the variances are equal). But this time this is a one sided t-test

In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file `files_for_lab/machine.txt`.
Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other




In [1]:
import pandas as pd
import numpy as np
import datetime
import warnings
import matplotlib.pyplot as plt
import seaborn as sns

warnings.filterwarnings('ignore')

In [38]:
data = pd.read_excel('machine.xlsx')
data

new_machine  old_machine
0         42.1         42.7
1         41.0         43.6
2         41.3         43.8
3         41.8         43.3
4         42.4         42.5
5         42.8         43.5
6         43.2         43.1
7         42.3         41.7
8         41.8         44.0
9         42.7         44.1

In [39]:
new = data["new_machine"]

In [40]:
old = data["old_machine"]

In [43]:
data.describe()

new_machine  old_machine
count    10.000000    10.000000
mean     42.140000    43.230000
std       0.683455     0.749889
min      41.000000    41.700000
25%      41.800000    42.800000
50%      42.200000    43.400000
75%      42.625000    43.750000
max      43.200000    44.100000

In [44]:
sample_mean_new = 42.14
sample_std_new = 0.683455
n1 = 10

sample_mean_old = 43.23
sample_std_old = 0.749889
n2 = 10

In [45]:
new = norm.rvs(loc=sample_mean_new, scale=sample_std_new, size=10)
old = norm.rvs(loc=sample_mean_old, scale=sample_std_old, size=10)

In [46]:
ttest_ind(new, old)

Ttest_indResult(statistic=-3.5405530260886073, pvalue=0.0023364835111191395)

- Degrees of freedom = (10-1)+(10-1) = 18 
- Significance level = 0.05 
- T-critical = 2.101

We can reject null hypothesis with 95% confidence. Old machine is significantly faster, because t-stat 3.54 > t_critical. 

## 2. An additional problem (not mandatory): In this case we can't assume that the population variances are equal. Hence in this case we cannot pool the variances.
   Independent random samples of 17 sophomores and 13 juniors attending a large university yield the following data on grade point averages. Data is provided in the file `files_for_lab/student_gpa.txt`.
   At the 5% significance level, do the data provide sufficient evidence to conclude that the mean GPAs of sophomores and juniors at the university differ?

   Test statistics can be calculated as: [link to the image - Test statistics calculation for Unpooled Variance Case](https://education-team-2020.s3-eu-west-1.amazonaws.com/data-analytics/7.04/7.04-unpooled_variances.png)

   Degrees of freedom is `(n1-1)+(n2-1)`.

In [16]:
data_1 = pd.read_excel('student_gpa.xlsx')
data_1

sophomores  juniors
0         3.04     2.56
1         1.71     2.77
2         3.30     2.70
3         2.88     3.00
4         2.11     2.98
5         2.60     3.47
6         2.92     3.26
7         3.60     3.20
8         2.28     3.19
9         2.82     2.65
10        3.03     3.00
11        3.12     3.39
12        2.86     2.58
13        3.49      NaN
14        3.11      NaN
15        2.13      NaN
16        3.27      NaN

In [17]:
sophomores = data_1["sophomores"]

In [22]:
juniors = [2.56, 2.77, 2.7, 3, 2.98, 3.47 , 3.26 , 3.2´,3.19 ,2.65, 3, 3.39,2.58]

In [47]:
data_1.describe()

sophomores    juniors
count   17.000000  13.000000
mean     2.839412   2.980769
std      0.519489   0.309259
min      1.710000   2.560000
25%      2.600000   2.700000
50%      2.920000   3.000000
75%      3.120000   3.200000
max      3.600000   3.470000

In [48]:
sample_mean_sophomores = 2.83
sample_std_sophomores = 0.51
n1 = 17

sample_mean_juniors = 2.98
sample_std_juniors = 0.30
n2 = 13

In [49]:
sophomores = norm.rvs(loc=sample_mean_sophomores, scale=sample_std_sophomores, size=n1)
juniors = norm.rvs(loc=sample_mean_juniors, scale=sample_std_juniors, size=n2)

In [50]:
ttest_ind(sophomores, juniors)

Ttest_indResult(statistic=1.0756662388516862, pvalue=0.29125830860530916)

- Degrees of freedom = (17-1)+(13-1) = 28
- Significance level = 0.05
- T-critical = 1.701

We can't reject null hypothesis with 95% confidenceMean of sophomores is not significantly different, because t-stat 1.07 < t_critical.